### PR TITLE
Configure Dependabot

### PR DESCRIPTION
**Pull request by gliptak**
Signed-off-by: Gábor Lipták <gliptak@gmail.com>

### CODE REVIEW

**Comment by gliptak** [comment]
@jandubois @ericpromislow please approve workflow

**Comment by ericpromislow** [comment]
Thanks for the PR. We're typically dealing with PRs on an oldest-first basis, and will get to this in due course.

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+  # Maintain dependencies for npm
+  - package-ecosystem: "npm"
+    directory: "/"
+    schedule:
+      interval: "weekly"
+
+  # Maintain dependencies for Golang
+  - package-ecosystem: "gomod"
+    directory: "/"
```
Comment (mook-as) [inline .github/dependabot.yml:18]: Isn't our go code all in `/src/go/…`? (There's two projects, `nerdctl-stub` and `wsl-helper`; does dependabot do the correct thing for both?)

**Review by mook-as — COMMENTED** [review]
Should we start out with a lower pull request limit (probably `1` per type, for now) as we start using it? We aim to keep the number of open PRs low, and given how out of date everything is we should start small to avoid a deluge of PRs.

We may also want to consider starting with `open-pull-requests-limit: 0` to _just_ do security updates first (and expand later).

Our test automation isn't at a point where I'd feel comfortable blinding bumping versions; we probably still need to do manual smoke testing to ensure things work. (For example, if we bump the electron-builder version too far I expect the update stuff to break.)

File: .github/dependabot.yml (hunk 1 above)
Comment (gliptak) [reply .github/dependabot.yml:18]: from what I understand `dependabot` finds packaging constructs and submit updates against them

**Comment by gliptak** [comment]
let me know if I'm to resubmit with `github-acttions` only. those PRs might be the most self-testing

**Review by mook-as — CHANGES_REQUESTED** [review]
I tried this out on my fork of the repo.

It found npm & GitHub Actions, but did _not_ find `go.mod`.  It showed an error:
```
Dependabot couldn't find a go.mod

Dependabot couldn't find a go.mod.

Dependabot requires a go.mod to evaluate your Go dependencies. It had expected to find one at the path: /go.mod.
```

Note that the `Dependencies` tab _did_ find things; I guess there's two completely separate flows?

**Review by mook-as — APPROVED** [review]
LGTM (though I still have worries of too many simultaneous PRs; we can always change it if it ends up not being what we want, though).

**Comment by jandubois** [comment]
@gliptak Why did you close this PR? We were still planning on merging it; we just wanted to get the 0.7.0 release out of the way.

Or did you just delete the branch by accident?

**Comment by gliptak** [comment]
@jandubois reopened as #1179

**Comment by jandubois** [comment]
Thank you!

**Pull request closed without merging**